lib/pure/random.nim

```diff
@@ -162,9 +165,17 @@ proc rand*[T](r: var Rand; a: openArray[T]): T {.deprecated.} =
   ## **Deprecated since v0.20.0:** use ``sample`` instead.
   result = a[rand(r, a.low..a.high)]
 
-proc rand*[T: SomeInteger](t: typedesc[T]): T =
-  ## Returns a random integer in the range `low(T)..high(T)`.
-  result = cast[T](state.next)
+proc rand*[T: SomeInteger](r: var Rand; t: typedesc[T]): T =
+  ## Returns a random element in the range `low(T)..high(T)`.
+  result = cast[T](r.next)
+
+proc rand*[T: enum](r: var Rand; t: typedesc[T]): T =
+  ## Returns a random element in the range `low(T)..high(T)`.
+  result = T(rand(r, ord(high(T))))
```

Comment:
This is still incorrect for enums that doesn't start at zero. `rand(r, low(T)..high(T))` will work except for enums with holes, but that might not matter.

Reply:
Maybe we should have a typetraits that test if enum has hole and statically fail for that case?

Reply:
Enums with holes are bad, we need to remove them from the language.